E4 Novice

Brian Bannon

The purpose of this analysis is to evaluate the quality of brake rotors produced during a one-week manufacturing run. By examining key measurements—thickness, diameter, surface roughness, and runout—we can identify which parts meet specifications, determine the yield for each measurement, and compare performance across production shifts. This information helps the quality control team detect process issues, reduce scrap, and maintain safety and reliability standards. Using Python and pandas, we will calculate statistics, assess yields, and provide actionable recommendations to improve production quality.

In [46]:
import pandas as pd

In [47]:
def calculate_basic_statistics(data: pd.DataFrame, measurement_column: str) -> dict:
    """
    Function to calculate basic statistics for a measurement column
    :param data: pandas DataFrame containing production measurements
    :param measurement_column: name of the column to analyze (e.g., 'Thickness_mm')
    :return: dictionary with keys 'mean', 'std', 'min', 'max' containing
             the corresponding statistical values
    """

    column = data[measurement_column]

    return {
        'mean': column.mean(),
        'std': column.std(),
        'min': column.min(),
        'max': column.max()
    }

In [48]:
def count_parts_in_spec(data: pd.DataFrame, measurement_column: str, lower_limit: float, upper_limit: float):
    """
    Function to count the number of parts within specification limits
    :param data: pandas DataFrame containing production measurements
    :param measurement_column: name of the column to check (e.g., 'Thickness_mm')
    :param lower_limit: lower specification limit for the measurement
    :param upper_limit: upper specification limit for the measurement
    :return: integer count of values within the specified limits (inclusive)
    """
    
    column = data[measurement_column]

    series = (column >= lower_limit) & (column <= upper_limit)

    return series.sum()

In [49]:
def analyze_by_shift(data: pd.DataFrame) -> pd.DataFrame:
    """
    Function to analyze production performance by work shift
    :param data: pandas DataFrame containing production measurements and a 'Shift' column
    :return: pandas DataFrame indexed by shift with average values for each measurement column
             ('Thickness_mm', 'Diameter_mm', 'Surface_Roughness_um', 'Runout_mm') and a count
             of parts produced per shift
    """
    measurement_columns = ['Thickness_mm', 'Diameter_mm', 'Surface_Roughness_um', 'Runout_mm']

    d = data.groupby('Shift')

    means = d[measurement_columns].mean()

    counts = d.size().rename('Count')

    results = means.join(counts)

    return results

In [50]:
data = pd.read_csv('E4_brake_rotor_production.csv')
calculate_basic_statistics(data, 'Thickness_mm')

{'mean': np.float64(12.013645200000001),
 'std': np.float64(0.060158001246573986),
 'min': np.float64(11.829),
 'max': np.float64(12.277)}

In [51]:
count_parts_in_spec(data, 'Thickness_mm', 11.95, 12.05)

np.int64(3160)

In [52]:
analyze_by_shift(data).round(3)

,Thickness_mm,Diameter_mm,Surface_Roughness_um,Runout_mm,Count
Shift,,,,,
Afternoon,12.011,303.130,1.291,0.019,1689
Morning,12.015,303.021,1.004,0.016,1780
Night,12.015,303.032,1.019,0.020,1531


Question 1  
Among the four measurements, Thickness_mm has the highest standard deviation relative to its mean (0.060 / 12 ≈ 0.5%). This indicates that rotor thickness varies more than other measurements, suggesting that the machining process for thickness might need tighter control or more frequent calibration.

Question 2  
For thickness, 3160 out of 5000 parts are in spec, giving a yield of 63.2%. Diameter, roughness, and runout would be similar calculations, but given the averages, diameter seems furthest from target (around 303 mm vs. 300 mm), likely resulting in the lowest yield. Low yield is critical because it increases scrap costs and reduces overall production efficiency.

Question 3  
Based on averages, the Morning shift produces the best-quality parts overall: it has the lowest surface roughness (1.004 µm), low runout (0.016 mm), and thickness closest to the target. This suggests that machines and operators on this shift maintain better process control.

Question 4  
The overall yield (parts meeting all four specifications) is lower than individual yields because a part must pass all criteria simultaneously. Even if a part meets thickness, it may fail diameter or surface roughness, compounding losses across measurements and lowering total acceptable output.

Question 5  
I would recommend focusing on controlling diameter, since it shows the largest deviation from target. Adjusting machine calibration, monitoring tool wear, and implementing in-process checks could reduce out-of-spec parts, improve yield, and save significant scrap costs.

In conclusion, the data analysis shows that rotor thickness exhibits the highest variation, while diameter consistently deviates most from the target, resulting in the lowest yield. The Morning shift produces the most consistent and high-quality parts across all measurements, suggesting effective machine operation and process control during that period. Overall yield is lower than individual measurement yields because parts must meet all specifications simultaneously. Focusing on improving diameter control and monitoring critical process parameters can increase yield, reduce scrap costs, and ensure safer, more reliable brake rotors.